lib/valueflow.cpp

```diff
@@ -769,9 +769,9 @@ static void valueFlowArrayElement(TokenList& tokenlist, const Settings& settings
                     const Token* arg = args[static_cast<std::size_t>(index)];
                     if (!arg->hasKnownIntValue())
                         continue;
-                    const ValueFlow::Value& v = arg->values().front();
-                    result.intvalue = v.intvalue;
-                    result.errorPath.insert(result.errorPath.end(), v.errorPath.cbegin(), v.errorPath.cend());
+                    const ValueFlow::Value* v = arg->getKnownValue(ValueFlow::Value::ValueType::INT);
```

Comment:
I don't have a strong opinon but well the previous `hasKnownIntValue()` check could be removed. and then we can check `v != nullptr` instead.. do it as you want..

Reply:
That check is indeed no longer necessary. The `getKnownValue()` changes were done after this change and I did not check the code if it could be simplified. Will do so.

Reply:
Done.